packages/theia-integration/src/browser/diagram/theia-marker-manager.ts

```diff
@@ -40,20 +42,52 @@ export function connectTheiaMarkerManager(
     }
 }
 
+type ValidationMarker = Marker & { reason?: string };
```

Comment:
Is there a particular reason why we don add the `reason` property directly to the `Marker` interface in the GLSP client packages?

Reply:
Yes, sort of... in my first version I even had it this way. But I opted to only mention the `reason` in the `SetMarkersAction` to make it more clear and easier to handle that each `SetMarkersAction` is meant to replace all markers that of a previous `SetMarkersAction` with the same `reason`.
If we'd put the `reason` in the `Marker` we may receive a set of markers with mixed reasons, which would complicate the handling on the client to remove the correct markers from previous `SetMarkersAction`.
But I agree, it is a bit cumbersome here that we need to remember it by extending the interface.